src/App.js

```diff
@@ -3,28 +3,55 @@ import './App.css';
 import LoginPage from './components/users/login/LoginPage';
 import RegistrationPage from './components/users/registration/RegistrationPage';
 import PostsPage from './components/posts/PostsPage';
-import DeliveryLogDash from './components/users/teachers/DeliveryLogDash'
+import DeliveryLogDash from './components/users/teachers/DeliveryLogDash';
 import { Routes, Route, Navigate, Outlet } from 'react-router-dom';
 import { loggedInUserContext } from './Helper/loggedInUserContext';
-
+import { useEffect } from 'react';
 import Profile from './components/profile/Profile';
-
+import client from './utils/client';
+import SearchBar from './components/searchBar/SearchBar';
+import RenderListOfStudents from './components/searchBar/RenderListOfStudents';
 
 function App() {
-  const [loggedInUser, setLoggedInUser] = useState(JSON.parse(localStorage.getItem('loggedInUser'))  );
-
-  return (
-    <loggedInUserContext.Provider value={{ loggedInUser, setLoggedInUser }}>
-      <div className='App'>
-        <Routes>
-          <Route path='/' element={<LoginPage />} />
-          <Route path='/signup' element={<RegistrationPage />} />
-          <Route element={<AuthenticateUser />}>
-            <Route path='/posts' element={<PostsPage />} />
+  const [userDataToRender ,setUserDataToRender] = useState({})
+  const [ userName, setUserName] = useState({
```

Comment:
It's not exactly clear what `userName` is here. If this is the user's name that we are searching for, it might be best if we name it something like `nameToSearch`